### PR TITLE
Add A/B test for marketing checkbox

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

This adds an A/B test to analyse the performance of the in-article email sign-up with the marketing checkbox. It uses legalese marketing text provided by marketing.
## What is the value of this and can you measure success?

This will help us review and analyse the impact of different scenarios:
- Marketing checkbox automatically checked (active un-check by user)
- Marketing checkbox unchecked (active check by user)
- Control - no checkbox

against the following metrics:
- Active close by user
- Sign-ups by user, with marketing actively checked
- Sign-ups by user, with marketing actively unchecked
- Sign-ups by user, with marketing left as default checked
- Sign-ups by user, with marketing left as default un-checked
## Screenshots

image
## CC

@amyhughes / @crifmulholland 

v37: `Article:rtrt | email form inline | article | 218 | user not signed-in | subscribe successful | marketing allowed`

`marketing allowed` or `marketing disallowed`

and v51 provides the ab test of `AB | EmailSignupMarketingCheckbox | marketing-default-checked` or `marketing-default-unchecked` or `control`

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/experiments/tests/email-signup-marketing-checkbox.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    Id
+) {
+    return function () {
+
+        this.id = 'EmailSignupMarketingCheckbox';
+        this.start = '2016-03-09';
+        this.expiry = '2016-03-18';
+        this.author = 'Gareth Trufitt';
+        this.description = 'Testing how the marketing checkbox affects engagement and sign-ups';
+        this.audience = 0.2;
```
Comment (crifmulholland) [inline static/src/javascripts/projects/common/modules/experiments/tests/email-signup-marketing-checkbox.js]: This is too high - can we push this down to 1% please

**Comment by rendro** [comment]
:+1:

File: common/app/views/fragments/email/signup/emailSignUp.scala.html [hunk 2]
```diff
@@ -29,6 +29,7 @@
             </div>
             <button type="submit" class="email-sub__submit-input js-email-sub__submit-input button button--tertiary button--large">@fragments.inlineSvg("envelope", "icon", Seq("submit-input__icon"))@submitText</button>
             <small class="email-sub__small js-email-sub__small">@smallText</small>
+            <label class="email-sub__small email-sub__small--mark email-sub__small--is-hidden js-email-sub__small--mark"><input type="checkbox" name="mark" class="email-sub__mark-input js-email-sub__mark-input"/></label>
```
Comment (sndrs) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: this adds an input, but with no non-js label – is that safe?

File: static/src/javascripts/projects/common/modules/email/email.js [hunk 3]
```diff
@@ -247,6 +265,17 @@ define([
                         $(el).addClass('email-sub--' + formModClass);
                     }
 
+                    if (showMarketingCheckbox && (showCheckedMarketing || showUncheckedMarketing)) {
```
Comment (sndrs) [inline static/src/javascripts/projects/common/modules/email/email.js]: fastdom?

File: common/app/views/fragments/email/signup/emailSignUp.scala.html (hunk 2 above)
Comment (gtrufitt) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: The whole thing is `display: none` (so hidden to visual and non-visual users) without JS as it's an A/B test (and not passed through to email tool at the moment) and shown as part of the A/B test. The label will also work with non-js too as it wraps the input.

File: static/src/javascripts/projects/common/modules/email/email.js (hunk 3 above)
Comment (gtrufitt) [inline static/src/javascripts/projects/common/modules/email/email.js]: Bit further up - https://github.com/guardian/frontend/pull/12240/files#diff-ff7248bf54a77d4c459fc27b774befd9R251

File: common/app/views/fragments/email/signup/emailSignUp.scala.html (hunk 2 above)
Comment (sndrs) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: what if there's no css? if it's only used in a JS context, could you insert it with js?

Comment (gtrufitt) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: Aye, good shout, will do.

Comment (sndrs) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: feels like there should either be a meaningful message you update or it should be added at the point the message is chosen?

Comment (gtrufitt) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: Yeah, I'll add this directly in the JS for the A/B test, if it moves into production then it'll be a part of the DOM with a meaningful message.

Comment (sndrs) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: oops sorry, your reply hadn't shown before my 3rd comment

Comment (gtrufitt) [inline common/app/views/fragments/email/signup/emailSignUp.scala.html]: No worries, that's updated now to be inserted in the A/B test check!

**Comment by sndrs** [comment]
is a response a :+1:? i don't know now. but this is

**Comment by gtrufitt** [comment]
I would say a :+1: on the main comment is a +1 :smile: Thanks!

**Pull request merged**